llvm/lib/Object/DXContainer.cpp

```diff
@@ -288,6 +290,51 @@ Error DirectX::RootSignature::parse(StringRef Data) {
                             llvm::Twine(FValue));
   Flags = FValue;
 
+  Current = Begin + RootParametersOffset;
+  for (uint32_t It = 0; It < NumParameters; It++) {
+    DirectX::RootParameter NewParam;
+
+    NewParam.Header.ParameterType =
+        support::endian::read<dxbc::RootParameterType,
+                              llvm::endianness::little>(Current);
+    if (!dxbc::RootSignatureValidations::isValidParameterType(
+            NewParam.Header.ParameterType))
```

Comment:
Doesn't this warn that we're converting to an enum value without an explicit cast? In any case, we have to check that the value is in range before converting to the enum type to avoid undefined behaviour ([[expr.static.cast] p10](https://timsong-cpp.github.io/cppwp/n4659/expr.static.cast#10))

Reply:
Moot now that the validations stuff isn't here any more.